### PR TITLE
feat: rest revamp

### PR DESCRIPTION
**Pull request by anwarulislam**
### Description
This PR introduces a new Tab feature for our Hoppscotch.io REST Platform. This Tab System has been long requested by our users and will greatly enhance their experience.

Before this feature, our platform lacked the ability to easily switch between different requests within a single view. With this new tab system, users can now navigate between multiple requests with ease, leading to a more organized and streamlined experience.

We are storing all tabs inside `RESTSession.ts` store. and each tab has `request` property containing request object for that tab.

### Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Review by nivedin — CHANGES_REQUESTED** [review]

File: packages/hoppscotch-common/src/components/http/RequestTab.vue [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+<script setup lang="ts">
+import { cloneDeep } from "lodash-es"
+import { ref, watch } from "vue"
+import { RESTTab } from "~/newstore/RESTSession"
+
+const props = defineProps<{ modelValue: RESTTab }>()
+const emit = defineEmits(["update:modelValue"])
+
+const tab = ref(cloneDeep(props.modelValue))
```
Comment (AndrewBastin) [inline packages/hoppscotch-common/src/components/http/RequestTab.vue:21]: Why the `cloneDeep` ?

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
please do look into the given issues.

**Review by nivedin — CHANGES_REQUESTED** [review]
@anwarulislam please look into these comments

File: packages/hoppscotch-common/src/pages/index.vue [hunk 2]
```diff
@@ -152,12 +176,107 @@ const onResolveConfirmSaveTab = () => {
...
+  // Load request on login resolve and start sync
+  onLoggedIn(async () => {
+    if (
+      Object.keys(route.query).length === 0 &&
+      !(route.query.code || route.query.error)
+    ) {
+      const tabStateFromSync =
+        await platform.sync.tabState.loadTabStateFromSync()
+
+      console.log("TABSTATE FROM SYNC", tabStateFromSync)
```
Comment (AndrewBastin) [inline packages/hoppscotch-common/src/pages/index.vue:261]: remove the console.log

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
only the current given issues. rest all looks good

**Review by nivedin — CHANGES_REQUESTED** [review]
@anwarulislam please look into these issues.

File: packages/hoppscotch-common/src/components/collections/index.vue
Comment (nivedin) [inline packages/hoppscotch-common/src/components/collections/index.vue:1]: when deleting a team request which is opened as an active tab throws this error

image

**Review by nivedin — CHANGES_REQUESTED** [review]
only found this issue.. rest looks all good

**Review by AndrewBastin — APPROVED** [review]
lgtm 💯

**Pull request merged**